ssl/t1_lib.c

```diff
@@ -2743,6 +2728,30 @@ int ssl_check_clienthello_tlsext_late(SSL *s)
     } else
         s->tlsext_status_expected = 0;
 
+    if (s->ctx->alpn_select_cb != NULL) {
```

Comment:
I'd suggest that you now make it `tls1_alpn_handle_client_hello_late` or some such. The times of 3000-line spaghetti methods are over :)

Also, please add a comment somewhere explaining that this order is due to SNI effects.